src/prefect/serialization/storage.py

```diff
@@ -40,6 +40,7 @@ class Meta:
         object_class = Azure
 
     container = fields.String(allow_none=False)
+    connection_string_secret_name = fields.String(allow_none=True)
```

Comment:
I am just wondering why this is needed here but S3Storage and GCSStorage don't?

Reply:
I think this is related because AWS_CREDENTIALS and GCP_CREDENTIALS are special default Secrets that Prefect looks for. Do we want to relevant AZURE_CREDENTIALS to the same status if possible where it becomes a default secret name?

Reply:
The auth process is a bit different for those cloud providers. We create Azure client object using [`from_connection_string` ](https://docs.microsoft.com/en-us/python/api/overview/azure/storage-blob-readme?view=azure-python#creating-the-client-from-a-connection-string) for Azure storage. Although we can create `AZURE_CREDENTIALS` dictionary with a key `AZURE_STORAGE_CONNECTION_STRING` set to a connection string, but I think it might be a bit confusing. 

User can accomplish the same behavior using:
* env variable on run config or agent: `AZURE_STORAGE_CONNECTION_STRING`
* save a secret with connection string in Prefect Cloud, and provide secret name in storage:
```
STORAGE = Azure(
    container="mariiacontainer",
    blob_name="test",
    connection_string_secret="MARIIA_SECRET_AZURE_CONNECTION",
)
```
* set a secret in context: `export PREFECT__CONTEXT__SECRETS__MYSECRET="MY SECRET VALUE"` or 
```
[context.secrets]
MYSECRET = "MY SECRET VALUE"
```
Let me know if you think we should create `AZURE_CREDENTIALS` dictionary with a key `AZURE_STORAGE_CONNECTION_STRING` as an option.

Reply:
The fallback would be ideal, but I think it's out of the scope of the very PR. You have a very nice writeup here though. Consider making an issue with it?